students/alexLaws/lesson04/mailroom.py

```diff
@@ -1,118 +1,92 @@
 #!/usr/bin/env python3
 
-donor_list = []
-donor_list.append({'Name': 'Ryan Moore', 'Donation': 500})
-donor_list.append({'Name': 'Ryan Moore', 'Donation': 250})
-donor_list.append({'Name': 'Ted Laws', 'Donation': 1000})
-donor_list.append({'Name': 'Ted Laws', 'Donation': 100})
-donor_list.append({'Name': 'Ben Snell', 'Donation': 150})
-donor_list.append({'Name': 'Andrew Crawford', 'Donation': 2000})
-donor_list.append({'Name': 'Andrew Crawford', 'Donation': 2000})
-donor_list.append({'Name': 'Andrew Crawford', 'Donation': 4000})
-donor_list.append({'Name': 'Beth Ross', 'Donation': 400})
-
-
-def user_action():
-    print("Choose an action from this list:")
-    print("Send a Thank You")
-    print("Create a Report")
-    print("Quit")
+import os
+
+donor_dictionary = {
+    'Ryan Moore': [500, 250], 'Ted Laws': [1000, 100], 'Ben Snell': [150],
+    'Andrew Crawford': [2000, 2000, 4000], 'Beth Ross': [400]}
+
+
+prompt = ("\nWhat would you like to do\n"
+          "Choose an action from this list:\n"
+          "1 - Send a Thank You\n"
+          "2 - Thank Everyone\n"
+          "3 - Create a Report\n"
+          "4 - Quit\n")
+
+
+def menu_selection(prompt, dispatch_dict):
     while True:
-        option = input("What is your choice: ")
-        if option.lower() in ["send a thank you", "create a report", "quit"]:
-            return option
+        response = input(prompt)
+        if dispatch_dict[response]() == "Exit Menu":
             break
-        else:
-            print("That is not an option. Choose again.")
 
 
-def dnrs(x):
-    names = []
-    for dnr in x:
-        person = dnr.get("Name")
-        if person in names:
-            pass
-        else:
-            names.append(person)
-    return names
+def exit():
+    return "Exit Menu"
 
 
-def send_to(x):
-    donors = dnrs(x)
-    recipient = input("Who is the thank you note for? Enter a full name or 'list' to see a list: ")
+def send_to():
+    recipient = input("Who is the thank you note for?\n"
+                      "Enter a full name or 'list' to see a list: ")
     if recipient.lower() == 'list':
         print("Here are all the donors:")
-        for name in donors:
-            print(name)
+        for i in donor_dictionary:
+            print(i)
```

Comment:
This can be written without `else` statement as:
```
    if recipient.lower() == 'list':
        print("Here are all the donors:")
        for i in donor_dictionary:
            print(i)
        recipient = input("Who is the thank you note for? ")
    return recipient
```